09-features.md

```diff
@@ -64,10 +64,14 @@ The origin node:
     unless indicated that it must set the odd feature bit instead.
   * MUST NOT set feature bits it does not support.
   * MUST NOT set feature bits in fields not specified by the table above.
+  * MUST NOT set both the optional and mandatory field.
   * MUST set all transitive feature dependencies.
+  * MUST support:
+    * `var_onion_optin`
 
-The origin node MUST support:
-  * `var_onion_optin`
+The receiving node:
+  * if both the optional and the mandatory feature its in a pair are set,
+  the feature should be treated as mandatory.
```

Comment:
```suggestion
  * if both the optional and the mandatory feature its in a pair are set:
    * the feature SHOULD be treated as mandatory
    * the feature MUST NOT be treated as optional
    * the receiving node MAY fail the feature negotiation
```

See [this](https://github.com/lightning/bolts/pull/1095/files#r1284884200) for rationale

Reply:
>the receiving node MAY fail the feature negotiation

No, this is the point of my PR. It is ok to declare both fields and if they are, the node should take the mandatory. This suggestion does not convince me, I am sorry.